packages/expect/src/matchers.js

```diff
@@ -423,6 +423,49 @@ const matchers: MatchersObject = {
     return {message, pass};
   },
 
+  async toHaveLengthAsync(received: any, lengthPromise: Promise<number>) {
```

Comment:
we don't want a new matcher, but you can add it to tests for integration testing

Reply:
Cool, so, we have integration tests were we can implement matchers to test stuff ?

Reply:
There are some in https://github.com/facebook/jest/blob/1eb02fd41ad6840350a5b2c0636151e2da8bcd8b/packages/expect/src/__tests__/extend.test.js, but just setting up a new one in https://github.com/facebook/jest/tree/master/integration-tests should be pretty straight forward if that's a better fit

Reply:
Awesome! I will use that as a reference! Will add some simple wrapper for `toHaveLength`